sdk/metricsadvisor/ai-metrics-advisor/review/ai-metrics-advisor.api.md

```diff
@@ -108,74 +117,123 @@ export type AnomalyValue = "AutoDetect" | "Anomaly" | "NotAnomaly";
     blobTemplate: string;
-    connectionString: string;
+    connectionString?: string;
     container: string;
 }
 
 // @public
 export type AzureCosmosDBDataFeedSource = {
     dataSourceType: "AzureCosmosDB";
-    dataSourceParameter: AzureCosmosDBParameter;
-};
+    authenticationType: "Basic";
+} & AzureCosmosDBParameter;
 
 // @public (undocumented)
 export interface AzureCosmosDBParameter {
     collectionId: string;
-    connectionString: string;
+    connectionString?: string;
     database: string;
     sqlQuery: string;
 }
 
+// @public
+export interface AzureDataExplorerAuthBasic {
+    // (undocumented)
+    authenticationType: "Basic";
+}
+
+// @public
+export interface AzureDataExplorerAuthManagedIdentity {
+    // (undocumented)
+    authenticationType: "ManagedIdentity";
+}
+
+// @public
+export interface AzureDataExplorerAuthServicePrincipal {
+    // (undocumented)
+    authenticationType: "ServicePrincipal";
+    // (undocumented)
+    credentialId: string;
+}
+
+// @public
+export interface AzureDataExplorerAuthServicePrincipalInKV {
+    // (undocumented)
+    authenticationType: "ServicePrincipalInKV";
+    // (undocumented)
+    credentialId: string;
+}
+
+// @public
+export type AzureDataExplorerAuthTypes = AzureDataExplorerAuthBasic | AzureDataExplorerAuthManagedIdentity | AzureDataExplorerAuthServicePrincipal | AzureDataExplorerAuthServicePrincipalInKV;
+
 // @public
 export type AzureDataExplorerDataFeedSource = {
     dataSourceType: "AzureDataExplorer";
-    dataSourceParameter: SqlSourceParameter;
-};
+    connectionString: string;
+    query: string;
+} & (AzureDataExplorerAuthTypes);
+
+// @public
+export type AzureDataLakeStorageGen2AuthTypes = DataLakeStorageGen2AuthBasic | DataLakeStorageGen2AuthManagedIdentity | DataLakeStorageGen2AuthServicePrincipal | DataLakeStorageGen2AuthServicePrincipalInKV | DataLakeStorageGen2AuthSharedKey;
 
 // @public
 export type AzureDataLakeStorageGen2DataFeedSource = {
     dataSourceType: "AzureDataLakeStorageGen2";
-    dataSourceParameter: AzureDataLakeStorageGen2Parameter;
-};
+    accountName: string;
+    fileSystemName: string;
+    directoryTemplate: string;
+    fileTemplate: string;
+} & (AzureDataLakeStorageGen2AuthTypes);
 
 // @public (undocumented)
 export interface AzureDataLakeStorageGen2Parameter {
     accountKey?: string;
-    accountName: string;
+    accountName?: string;
     directoryTemplate: string;
     fileSystemName: string;
     fileTemplate: string;
 }
 
+// @public (undocumented)
+export interface AzureSQLConnectionStringParam {
+    connectionString: string;
+}
+
 // @public
 export type AzureTableDataFeedSource = {
     dataSourceType: "AzureTable";
-    dataSourceParameter: AzureTableParameter;
+    connectionString: string;
+    table: string;
+    query: string;
+    authenticationType: "Basic";
 };
 
 // @public (undocumented)
 export interface AzureTableParameter {
-    connectionString: string;
+    connectionString?: string;
```

Comment:
do we still need this type? Looks that we already inline it above.

Reply:
Removed from exports